src/job_cache/job_cache_impl_common.cpp

```diff
@@ -251,11 +255,17 @@ void remove_backing_files(const std::string &dir, int64_t job_id) {
           .urgent()();
       exit(1);
     }
-    unlink_no_fail(entry->name.c_str());
+    files_to_remove.push_back(wcl::join_paths(job_dir, entry->name));
+  }
+  for (const auto &file : files_to_remove) {
+    unlink_no_fail(file.c_str());
   }
+
+  rmdir_no_fail(job_dir.c_str());
 }
```

Comment:
do you need to rebase?

Reply:
Nope! I found bugs in the removal while writing this change. This fixes them.